cmd/util/ledger/migrations/account_based_migration.go

```diff
@@ -0,0 +1,210 @@
+package migrations
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/pathfinder"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/util"
+)
+
+// PayloadToAccount takes a payload and return:
+// - (address, true, nil) if the payload is for an account, the account address is returned
+// - ("", false, nil) if the payload is not for an account
+// - ("", false, err) if running into any exception
+func PayloadToAccount(p ledger.Payload) (string, bool, error) {
+	k, err := p.Key()
+	if err != nil {
+		return "", false, fmt.Errorf("could not find key for payload: %w", err)
+	}
+	id, err := KeyToRegisterID(k)
+	if err != nil {
+		return "", false, fmt.Errorf("error converting key to register ID")
+	}
+	if len([]byte(id.Owner)) != flow.AddressLength {
+		return "", false, nil
```

Comment:
is this check safe? if i remember correctly address "" is a reserved address.

Reply:
there are registers with "" as an address, but those are not for a special address they are for special registers that don't belong to any address. I think this check is ok, but might be nicer packaged in a function with a clear name.